lldb/examples/python/parsed_cmd.py

```diff
@@ -0,0 +1,313 @@
+"""
+This module implements a couple of utility classes to make writing
+lldb parsed commands more Pythonic.
+The way to use it is to make a class for your command that inherits from ParsedCommandBase.
+That will make an LLDBOVParser which you will use for your
+option definition, and to fetch option values for the current invocation
+of your command.  Access to the OV parser is through:
+
+ParsedCommandBase.get_parser()
+
+Next, implement setup_command_definition() in your new command class, and call:
+
+  self.get_parser().add_option()
+
+to add all your options.  The order doesn't matter for options, lldb will sort them
+alphabetically for you when it prints help.
+
+Similarly you can define the arguments with:
+
+  self.get_parser().add_argument()
+
+At present, lldb doesn't do as much work as it should verifying arguments, it
+only checks that commands that take no arguments don't get passed arguments.
+
+Then implement the execute function for your command as:
+
+    def __call__(self, debugger, args_list, exe_ctx, result):
+
+The arguments will be a list of strings.  
+
+You can access the option values using the 'dest' string you passed in when defining the option.
+
+If you need to know whether a given option was set by the user or not, you can
+use the was_set API.
+
+There are example commands in the lldb testsuite at:
+
+llvm-project/lldb/test/API/commands/command/script/add/test_commands.py
+"""
+import inspect
+import lldb
+import sys
+from abc import abstractmethod
+
+class LLDBOVParser:
```

Comment:
I think I said this before, but I would just expand the class name to `LLDBOptionValueParser` (despite how Java enterprise ready (tm) that sounds).

It'll be read many more times than it's written and this example file is already enough to think about without chasing down what this acronym means. If the user's code is going to be littered with the literal name, then maybe keep it short but otherwise it's just one more unknown for them.

Reply:
`self.ov_parser` is fine as a variable name since any IDE helper will show this type name for it.

Reply:
Yes, I agree, nobody will ever type this one because that's not even the class you override.  So there's no reason for it to have a mysterious name.